src/pages/index.js

```diff
@@ -64,50 +88,151 @@ profileEditPopup.setEventListeners();
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({ name: userData.name, about: userData.about });
+    userInfo.setUserAvatar({ avatar: userData.avatar });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 profileEditButton.addEventListener("click", function () {
-  profileTitleInput.value = profileTitle.textContent.trim();
-  profileDescriptionInput.value = profileDescription.textContent.trim();
-  editFormValidator.resetValidation();
+  const userData = userInfo.getUserInfo();
+  profileTitleInput.value = userData.name;
+  profileDescriptionInput.value = userData.job;
   profileEditPopup.open();
 });
 
 cardAddButton.addEventListener("click", function () {
   newCardPopup.open();
 });
 
+// avatarPopup
+
+const avatarPopup = new PopupWithForm(
+  "#profile-picture-modal",
+  handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
+avatarEditButton.addEventListener("click", function () {
+  avatarPopup.open();
+});
+
 // Event Handlers
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  popupInstance.setLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.setLoading(false);
+    });
+}
 
 function handleImageClick({ name, link }) {
   imagePreviewPopup.open({ name, link });
 }
 
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo({ name: data.name, job: data.description });
-  imagePreviewPopup.close();
+  function makeRequest() {
+    return api.updateUserInfo(data).then((res) => {
+      userInfo.setUserInfo({ name: res.name, about: res.about });
+    });
+  }
+  handleSubmit(makeRequest, profileEditPopup);
 }
 
 function handleAddCardFormSubmit(inputValues) {
   const name = inputValues.name;
   const link = inputValues.link;
-  const cardData = { name: name, link: link };
-  renderCard(cardData);
-  newCardPopup.close();
-  cardFormElement.reset();
-  addCardFormValidator.disableButton();
+  function makeRequest() {
+    return api.addCard({ name, link }).then((cardData) => {
+      renderCard(cardData);
+      cardFormElement.reset();
+      addCardFormValidator.disableButton();
+    });
+  }
+  handleSubmit(makeRequest, newCardPopup);
+}
+
+function handleAvatarFormSubmit(userData) {
+  function makeRequest() {
+    return api.updateUserAvatar(userData).then((res) => {
+      userInfo.setUserAvatar({ avatar: res.avatar });
+    });
+  }
+  handleSubmit(makeRequest, avatarPopup);
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardclick,
+    handleLikeCard,
+    handleUnLikeCard
+  );
   return card.getView();
 }
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
+
+function handleDeleteCardclick(card) {
```

Comment:
`handleDeleteCardclick`


Every variable in the project should be in `camelCase` 


`handleDeleteCardClick`

Reply:
Actually, I meant this `camelCase` https://snipboard.io/2IkaOp.jpg